e2e/tests/core/02-client/client_test.go

```diff
@@ -130,6 +131,35 @@ func (s *ClientTestSuite) TestScheduleIBCUpgrade_Succeeds() {
 		s.Require().Equal("upgrade-client", plan.Name)
 		s.Require().Equal(planHeight, plan.Height)
 	})
+
+	t.Run("ensure legacy proposal does not succeed", func(t *testing.T) {
+
+		authority, err := s.QueryModuleAccountAddress(ctx, govtypes.ModuleName, chainA)
+		s.Require().NoError(err)
+		s.Require().NotNil(authority)
+
+		clientState, err := s.QueryClientState(ctx, chainB, ibctesting.FirstClientID)
+		s.Require().NoError(err)
+
+		originalChainID := clientState.(*ibctm.ClientState).ChainId
+		revisionNumber := clienttypes.ParseChainID(originalChainID)
+		// increment revision number even with new chain ID to prevent loss of misbehaviour detection support
+		newChainID, err = clienttypes.SetRevisionNumber(originalChainID, revisionNumber+1)
+		s.Require().NoError(err)
+		s.Require().NotEqual(originalChainID, newChainID)
+
+		upgradedClientState := clientState.ZeroCustomFields().(*ibctm.ClientState)
+		upgradedClientState.ChainId = newChainID
+
+		legacyUpgradeProposal, err := clienttypes.NewUpgradeProposal(ibctesting.Title, ibctesting.Description, upgradetypes.Plan{
+			Name:   "upgrade-client-legacy",
+			Height: legacyPlanHeight,
+		}, upgradedClientState)
+
+		s.Require().NoError(err)
+		txResp := s.ExecuteGovV1Beta1Proposal(ctx, chainA, chainAWallet, legacyUpgradeProposal)
+		s.AssertTxFailure(txResp, govtypes.ErrInvalidProposalContent)
```

Comment:
Interesting, I was expecting this proposal type to fail after the voting period passes. Do we know why this fails on submission?

Reply:
discussed on slack but can put the info here:

we hit [this](https://github.com/cosmos/cosmos-sdk/blob/6615ff4f76088d4329ebe71230f2f9ac522dcd8a/x/gov/keeper/proposal.go#L79-L84) error . We don't hit the `ErrNoProposalHandlerExists` because a handler [does exist](https://github.com/cosmos/ibc-go/blob/32560f4a765e990c34481422873a4765df4d61ca/modules/core/02-client/proposal_handler.go#L18-L28).

But we still error out.